packages/jazz-tools/src/coValues/coMap.ts

```diff
@@ -485,6 +489,77 @@ export class CoMap extends CoValueBase implements CoValue {
         }
         return this;
     }
+
+    asPlainData<M extends CoMap>(this: M): RecursiveCoMapInit<M>;
+    asPlainData<M extends CoMap, Depth>(
+        this: M,
+        depth: Depth & DepthsIn<M>,
+    ): Promise<RecursiveCoMapInit<M> | undefined>;
+    asPlainData<M extends CoMap, Depth>(
+        this: M,
+        depthOrSeen?:
+            | (Depth & DepthsIn<M>)
+            | WeakMap<CoMap | CoList, RecursiveCoMapInit<M>>,
+    ): RecursiveCoMapInit<M> | Promise<RecursiveCoMapInit<M> | undefined> {
+        if (depthOrSeen instanceof WeakMap) {
+            return this.asPlainDataSync(depthOrSeen);
+        } else if (depthOrSeen === undefined) {
+            return this.asPlainDataSync(new WeakMap());
+        } else {
+            return this.asPlainDataAsync(depthOrSeen, new WeakMap());
+        }
+    }
+
+    private asPlainDataSync<M extends CoMap>(
+        this: M,
+        seen: WeakMap<CoMap | CoList, RecursiveCoMapInit<M>>,
+    ): RecursiveCoMapInit<M> {
+        if (seen.has(this)) {
+            return seen.get(this) as RecursiveCoMapInit<M>;
+        }
+
+        const plainObject = {} as RecursiveCoMapInit<M>;
+        seen.set(this, plainObject);
+
+        for (const key of Object.keys(this)) {
+            const value = this[key as keyof M];
+            (plainObject as any)[key] = plainDataStrategy.toPlainData(
+                value as CoMap,
+                seen,
+            );
```

Comment:
**suggestion:** you can cast the type at his source making it easier to use `key` 

```
        for (const key of Object.keys(this) as Array<keyof M>) {
           const value = this[key] as CoMap;
           plainObject[key] = plainDataStrategy.toPlainData(
                value,
                seen,
          );
```